src/lavinmq/amqp/client.cr

```diff
@@ -97,15 +97,15 @@ module LavinMQ
       end
 
       private def read_loop
-        i = 0
+        delivered_bytes = 0_i32
         socket = @socket
         loop do
           AMQP::Frame.from_io(socket) do |frame|
             {% unless flag?(:release) %}
               @log.trace { "Received #{frame.inspect}" }
             {% end %}
-            if (i += 1) == 8192
-              i = 0
+            if (delivered_bytes &+= frame.bytesize) > 131_072 # 8192 * 16
```

Comment:
Just curious, would Crystal's compiler do the `8192 * 16` for you? if so I guess you could leave it as that and skip it as a comment :) I assume you wanted to avoid the multiplication for each iteration

Reply:
oh, I assume the compiler would do that multiplication just once anyway I think 🤔 
But I wanted it to be as clear as possible as to after how many bytes we yield. The comment is more of a clarification of why I selected the number 131_072 (because the smaller frames seem to be around 14-15 bytes, so 16 * 8192 makes some sense)

Reply:
Ah roger, thanks!

Reply:
Just because I got curious 🤓 
Seems very similar performance wise! 

```
def mult_each()
  i = 0
  loop do
    break if (i &+= 1) > 8192 * 16
  end
end

def consta()
  i = 0
  loop do
    break if (i &+= 1) > 131_072
  end
end
```

```
viktor@viktor-lenovo:~/code/_crystal-test$ crystal run bench.cr --release
mult_each 911.09M (  1.10ns) (± 4.20%)  0.0B/op        fastest
   consta 908.26M (  1.10ns) (± 5.03%)  0.0B/op   1.00× slower
```